sqlserver/datadog_checks/sqlserver/xe_collection/base.py

```diff
@@ -0,0 +1,910 @@
+            f"collect_raw_query={self._collect_raw_query}"
+        )
+
+        super(XESessionBase, self).__init__(
+            check,
+            run_sync=True,
+            enabled=True,
+            min_collection_interval=self._config.min_collection_interval,
+            dbms="sqlserver",
+            rate_limit=1 / float(self.collection_interval),
+            job_name=f"xe_{session_name}",
+            shutdown_callback=self._close_db_conn,
+        )
+        self._conn_key_prefix = f"dbm-xe-{session_name}-"
+        self._is_azure_sql_database = False
+        self._check_azure_status()
+
+    # Methods to allow subclasses to extend field definitions
+    def get_numeric_fields(self, event_type=None):
+        """Get numeric fields with defaults for given event type"""
+        return self.BASE_NUMERIC_FIELDS.copy()
+
+    def get_string_fields(self, event_type=None):
+        """Get string fields for given event type"""
+        return self.BASE_STRING_FIELDS.copy()
+
+    def get_sql_fields(self, event_type=None):
+        """Get SQL fields for given event type"""
+        if event_type == "sql_batch_completed":
+            return ["batch_text", "sql_text"]
+        elif event_type == "rpc_completed":
+            return ["statement", "sql_text"]
+        elif event_type == "module_end":
+            return ["statement", "sql_text"]
+        return self.BASE_SQL_FIELDS.copy()
+
+    def register_event_handler(self, event_name, handler_method):
+        """Register a handler method for a specific event type"""
+        self._event_handlers[event_name] = handler_method
+
+    def _check_azure_status(self):
+        """Check if this is Azure SQL Database"""
+        engine_edition = self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, "")
+        self._is_azure_sql_database = is_azure_sql_database(engine_edition)
+
+    def _close_db_conn(self):
+        """Close database connection on shutdown"""
+        pass
+
+    def session_exists(self):
+        """Check if this XE session exists and is running"""
+        with self._check.connection.open_managed_default_connection(key_prefix=self._conn_key_prefix):
+            with self._check.connection.get_managed_cursor(key_prefix=self._conn_key_prefix) as cursor:
+                # For Azure SQL Database support
+                level = ""
+                if self._is_azure_sql_database:
+                    level = "database_"
+
+                # Build the query with proper parameterization
+                query = f"SELECT 1 FROM sys.dm_xe_{level}sessions WHERE name = ?"
+                cursor.execute(query, (self.session_name,))
+
+                return cursor.fetchone() is not None
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _query_ring_buffer(self):
+        """
+        Query the ring buffer data and parse the XML on the client side.
+        This avoids expensive server-side XML parsing for better performance.
```

Comment:
What impact does this have on agent performance? Generally the agent is running on much smaller hardware than the database so if the performance hit is that severe will it cause problems with the agent?